doc/api/training/smp_versions/v1.10.0/smd_model_parallel_common_api.rst

```diff
@@ -0,0 +1,538 @@
+      the rank of the process across all processes used in training.
+   -  ``RankType.MP_RANK``: The associated rank is to be interpreted as the
+      rank of the process within the ``MP_GROUP``.
+   -  ``RankType.DP_RANK``: The associated rank is to be interpreted as the
+      rank of the process within the ``DP_GROUP``.
+
+
+**Communication primitives:**
+
+.. function:: smp.broadcast(obj, group)
+   :noindex:
+
+   Sends the object to all processes in the
+   group. The receiving process must call ``smp.recv_from`` to receive the
+   sent object.
+
+   **Inputs**
+
+   -  ``obj``: An arbitrary picklable Python object that will be broadcast.
+
+   -  ``group``: A ``CommGroup`` argument that represents to which group of
+      processes the object will be sent.
+
+   **Notes**
+
+   -  When you use ``broadcast`` on the sender process, there needs
+      to be an accompanying ``smp.recv_from()`` call on the receiver
+      processes.
+
+   -  This is a synchronous call; the ``broadcast`` statement
+      returns only after all ranks participating in the call have made a
+      matching ``recv_from`` call.
+
+   **Example**
+
+   .. code:: python
+
+      if smp.rank() == 0:
+          smp.broadcast(something, group=smp.CommGroup.WORLD)
+      else:
+          smp.recv_from(0, rank_type=smp.RankType.WORLD_RANK)
+
+.. function:: smp.send(obj, dest_rank, rank_type)
+   :noindex:
+
+   Sends the object ``obj`` to
+   ``dest_rank``, which is of a type specified by ``rank_type``.
+
+   **Inputs**
+
+   -  ``obj``: An arbitrary picklable Python object that will be sent.
+
+   -  ``dest_rank`` (``int``): An integer denoting the rank of the receiving process.
+
+   -  ``rank_type`` (``enum``): A ``smp.RankType`` ``enum`` that determines how
+      ``dest_rank`` is to be interpreted. For example if ``dest_rank`` is 1
+      and ``rank_type`` is ``MP_RANK``, then ``obj`` is sent to process
+      with ``mp_rank`` 1 in the ``MP_GROUP`` which contains the current
+      process.
+
+   **Notes**
+
+   -  Note: \ This is a synchronous call; the ``send`` statement returns
+      only after the destination rank has made a matching
+      ``recv_from`` call.
+
+.. function:: smp.recv_from(src_rank, rank_type)
+   :noindex:
+
+   Receive an object from a peer process. Can be used with a matching
+   ``smp.send`` or a ``smp.broadcast`` call.
+
+   **Inputs**
+
+   -  ``src_rank`` (``int``): An integer denoting rank of the sending process.
+
+   -  ``rank_type`` (``enum``): A ``smp.RankType`` ``enum`` that determines how
+      ``dest_rank`` is to be interpreted. For example if ``src_rank`` is 1
+      and ``rank_type`` is ``MP_RANK``, then the object is received from
+      the process with ``mp_rank`` 1 in the ``MP_GROUP`` which contains the
+      current process.
+
+   **Returns**
+
+   Returns the python object that is sent by the peer process.
+
+   **Notes**
+
+   -  Note: This is a synchronous call; the ``recv_from`` statement returns
```

Comment:
Curious why so many uses of escaping spaces, but not used here as in line 440. What formatting issues are you avoiding?

Reply:
It is actually to not put any space between `i` and `th`. After the code formatting, `\ ` should be added, otherwise the doc build breaks.